libs/core/algorithms/include/hpx/parallel/algorithms/detail/set_operation.hpp

```diff
@@ -74,11 +74,18 @@ namespace hpx::parallel::detail {
 
     struct set_chunk_data
     {
-        std::size_t start = static_cast<std::size_t>(-1);
-        std::size_t len = static_cast<std::size_t>(-1);
-        std::size_t start_index = static_cast<std::size_t>(-1);
-        std::size_t first1 = static_cast<std::size_t>(-1);
-        std::size_t first2 = static_cast<std::size_t>(-1);
+        static const std::size_t uninit_start = static_cast<std::size_t>(-1);
+        static const std::size_t uninit_len = static_cast<std::size_t>(0);
+        static const std::size_t uninit_start_index =
+            static_cast<std::size_t>(-1);
+        static const std::size_t uninit_first1 = static_cast<std::size_t>(-1);
+        static const std::size_t uninit_first2 = static_cast<std::size_t>(-1);
+
+        std::size_t start = uninit_start;
+        std::size_t len = uninit_len;
+        std::size_t start_index = uninit_start_index;
+        std::size_t first1 = uninit_first1;
+        std::size_t first2 = uninit_first2;
```

Comment:
What is the rationale for this change?

Reply:
After the async execution, we have an if conditional which checks if the chunk values are same an default values, in which case it does not process the chunk. I felt defining the default values was a better alternative to simply checking is start_index == -1. Note that the bug was because a default value was incorrect.

Reply:
Hmmm. What change fixed the issue? From what I can see, this PR does not introduce any changes that would modify behavior. I might miss something, however.

Reply:
len used to default to -1, similar to remaining variables. It is supposed to default to 0. There was also a out of bounds memory access I believe.

Reply:
Ok, now I see the change. If you want to separate the initial values from the initialization, I'd suggest using `static constexpr` variables, however.

Reply:
updated, missed that point earlier.